dds/DCPS/transport/framework/TransportSendBuffer.cpp

```diff
@@ -377,24 +377,34 @@ SingleSendBuffer::resend_fragments_i(const SequenceNumber& seq,
   if (fragments_.empty() || requested_frags.empty()) {
     return;
   }
-  const BufferMap& buffers = fragments_[seq];
-  const OPENDDS_VECTOR(SequenceRange) psr =
-    requested_frags.present_sequence_ranges();
-  SequenceNumber sent = SequenceNumber::ZERO();
-  for (size_t i = 0; i < psr.size(); ++i) {
-    BufferMap::const_iterator it = buffers.lower_bound(psr[i].first);
-    if (it == buffers.end()) {
-      return;
-    }
-    BufferMap::const_iterator it2 = buffers.lower_bound(psr[i].second);
-    while (true) {
-      if (sent < it->first) {
-        resend_one(it->second);
-        sent = it->first;
-      }
-      if (it == it2) {
-        break;
+  const FragmentMap::const_iterator fm_it = fragments_.find(seq);
+  if (fm_it == fragments_.end()) {
+    return;
+  }
+  const BufferMap& buffers = fm_it->second;
+  const OPENDDS_VECTOR(SequenceRange)& psr = requested_frags.present_sequence_ranges();
+
+  SequenceNumber frag_min;
+  size_t i = 0;
+  BufferMap::const_iterator it = buffers.lower_bound(psr.front().first);
+  BufferMap::const_iterator end = buffers.lower_bound(psr.back().second);
+  if (end != buffers.end()) {
+    ++end;
+  }
+
+  // Iterate over both containers simultaneously
+  while (i < psr.size() && it != end) {
+    if (psr[i].second < frag_min) {
+      // While the range is too low, increment range
```

Comment:
I think this is clear without the comment.  If you do want to keep the comment, consider revising it since it's not incrementing the range, but advancing to the next range in the collection